{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gerar_dados(tempo_seg):\n",
    "    dados = {\n",
    "        'data': [],\n",
    "        'pedidos_realizados': [],\n",
    "        'acumulado_pedidos_realizados': [],\n",
    "        'pedido_em_execucao': [],\n",
    "        'acumulado_pedidos_finalizados': []\n",
    "    }\n",
    "    data = datetime.datetime(2022, 1, 16, 8, 0, 0) # Ano, mês, dia, hora, minuto, segundo\n",
    "    pedidosRealizados = 0\n",
    "    pedidosFinalizados = 0  # Começa \n",
    "    aux_realizacao = 0\n",
    "    aux_execucao = 0\n",
    "    aux_finalizacao = 0\n",
    "    cont_realizados = 0\n",
    "    cont_execucao = 0\n",
    "    cont_finalizados = 0\n",
    "    execucao = False\n",
    "    for seg in range(1, tempo_seg + 1):\n",
    "        \n",
    "        #Limita a dia de semana\n",
    "        if data.weekday() >= 5:\n",
    "            tempoVoltar = 7 - data.weekday()\n",
    "            data += datetime.timedelta(days=tempoVoltar)\n",
    "\n",
    "        #Limita ao horario de 8 as 12 13 as 17\n",
    "        if data.hour >= 12 and data.hour <= 13:\n",
    "            data += datetime.timedelta(hours=1)\n",
    "        if data.hour > 17:\n",
    "            data += datetime.timedelta(hours=15)\n",
    "\n",
    "        #Limitações logicas sobre entrar em falha e conserto\n",
    "\n",
    "        aux_realizacao = random.randint(0,9)\n",
    "        if aux_realizacao == 1:\n",
    "            pedidosRealizados += 1\n",
    "            cont_realizados += 1\n",
    "            #pedidoFinalizado \n",
    "            #cont_realizado += 1\n",
    "\n",
    "        if pedidosRealizados > 0 and execucao == False:\n",
    "            aux_execucao = random.randint(0,2)\n",
    "            if aux_execucao == 1 or aux_execucao == 2:\n",
    "                execucao = True\n",
    "\n",
    "        if execucao == True:\n",
    "            if cont_execucao < 12:\n",
    "                cont_execucao += 1\n",
    "\n",
    "                # Adiciona os dados para a linha atual no df\n",
    "                dados['data'].append(data)\n",
    "                dados['pedidos_realizados'].append(pedidosRealizados)\n",
    "                dados['acumulado_pedidos_realizados'].append(cont_realizados)\n",
    "                dados['pedido_em_execucao'].append(execucao)\n",
    "                dados['acumulado_pedidos_finalizados'].append(cont_finalizados)\n",
    "                data += datetime.timedelta(seconds=15)\n",
    "\n",
    "                continue\n",
    "\n",
    "            else:\n",
    "                aux_finalizacao = random.randint(0,2)\n",
    "                if aux_finalizacao == 1 or aux_finalizacao == 2:\n",
    "                    cont_execucao = 0\n",
    "                    pedidosRealizados -= 1\n",
    "                    cont_finalizados += 1\n",
    "                    execucao = False\n",
    "\n",
    "        # Adiciona os dados para a linha atual no df\n",
    "        dados['data'].append(data)\n",
    "        dados['pedidos_realizados'].append(pedidosRealizados)\n",
    "        dados['acumulado_pedidos_realizados'].append(cont_realizados)\n",
    "        dados['pedido_em_execucao'].append(execucao)\n",
    "        dados['acumulado_pedidos_finalizados'].append(cont_finalizados)\n",
    "        data += datetime.timedelta(seconds=15)\n",
    "\n",
    "    return pd.DataFrame(dados)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "qtdGerada = 1000000  # Quantidade de dados a gerar por linha, intervalo entre linhas é 15 segundos."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "dados_pedidos = gerar_dados(qtdGerada)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dados_pedidos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "dados_pedidos.to_csv('dados_pedidos.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "dados_pedidos[\"data_abreviada\"] = dados_pedidos['data'].dt.strftime('%Y-%m-%d')\n",
    "dados_pedidos[\"data_abreviada\"] =  pd.to_datetime(dados_pedidos[\"data_abreviada\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\3285628536.py:13: RuntimeWarning: invalid value encountered in divide\n",
      "  (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n"
     ]
    }
   ],
   "source": [
    "dados_pedidos['taxa_de_saida(und/h)'] = ''\n",
    "anos = dados_pedidos['data_abreviada'].dt.year.unique()\n",
    "meses = dados_pedidos['data_abreviada'].dt.month.unique()\n",
    "dias = dados_pedidos['data_abreviada'].dt.day.unique()\n",
    "\n",
    "for ano in anos: \n",
    "    for mes in meses:\n",
    "        for dia in dias:\n",
    "            df_aux = dados_pedidos[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) & (dados_pedidos['data'].dt.day == dia)]\n",
    "            try:\n",
    "                dados_pedidos.loc[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) \\\n",
    "                            & (dados_pedidos['data'].dt.day == dia), \"taxa_de_saida(und/h)\"] = \\\n",
    "                                (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
    "                                    )/((df_aux.loc[df_aux['pedido_em_execucao'] == True, 'data'].max() - \\\n",
    "                                    df_aux.loc[df_aux['pedido_em_execucao'] == True,'data'].min()).total_seconds()/3600)\n",
    "            except ZeroDivisionError:\n",
    "                 dados_pedidos.loc[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) \\\n",
    "                            & (dados_pedidos['data'].dt.day == dia), \"taxa_de_saida(und/h)\"] = \\\n",
    "                                (df_aux.loc[:, 'acumulado_pedidos_finalizados'].max() - df_aux.loc[:, 'acumulado_pedidos_finalizados'].min()\\\n",
    "                                    )/0.00416666666666666666666666666667\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\1044519432.py:14: RuntimeWarning: invalid value encountered in divide\n",
      "  ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n"
     ]
    }
   ],
   "source": [
    "dados_pedidos['takt_time(s/und)'] = ''\n",
    "\n",
    "anos = dados_pedidos['data_abreviada'].dt.year.unique()\n",
    "meses = dados_pedidos['data_abreviada'].dt.month.unique()\n",
    "dias = dados_pedidos['data_abreviada'].dt.day.unique()\n",
    "\n",
    "for ano in anos: \n",
    "    for mes in meses:\n",
    "        for dia in dias:\n",
    "            df_aux = dados_pedidos[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) & (dados_pedidos['data'].dt.day == dia)]\n",
    "            try:\n",
    "                dados_pedidos.loc[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) \\\n",
    "                            & (dados_pedidos['data'].dt.day == dia), \"takt_time(s/und)\"] = \\\n",
    "                            ((df_aux.loc[:, 'data'].max() - df_aux.loc[:,'data'].min()).total_seconds())/\\\n",
    "                                (df_aux.loc[:, 'acumulado_pedidos_realizados'].max() - df_aux.loc[:, 'acumulado_pedidos_realizados'].min())\n",
    "            except ZeroDivisionError:\n",
    "                 dados_pedidos.loc[(dados_pedidos['data'].dt.year == ano) & (dados_pedidos['data'].dt.month == mes) \\\n",
    "                            & (dados_pedidos['data'].dt.day == dia), \"takt_time(s/und)\"] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data</th>\n",
       "      <th>pedidos_realizados</th>\n",
       "      <th>acumulado_pedidos_realizados</th>\n",
       "      <th>pedido_em_execucao</th>\n",
       "      <th>acumulado_pedidos_finalizados</th>\n",
       "      <th>data_abreviada</th>\n",
       "      <th>taxa_de_saida(und/h)</th>\n",
       "      <th>takt_time(s/und)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2022-01-17 08:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2022-01-17 08:00:15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2022-01-17 08:00:30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2022-01-17 08:00:45</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2022-01-17 08:01:00</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999995</th>\n",
       "      <td>2024-04-29 09:38:45</td>\n",
       "      <td>28338</td>\n",
       "      <td>99772</td>\n",
       "      <td>True</td>\n",
       "      <td>71434</td>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999996</th>\n",
       "      <td>2024-04-29 09:39:00</td>\n",
       "      <td>28339</td>\n",
       "      <td>99773</td>\n",
       "      <td>True</td>\n",
       "      <td>71434</td>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999997</th>\n",
       "      <td>2024-04-29 09:39:15</td>\n",
       "      <td>28339</td>\n",
       "      <td>99773</td>\n",
       "      <td>True</td>\n",
       "      <td>71434</td>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999998</th>\n",
       "      <td>2024-04-29 09:39:30</td>\n",
       "      <td>28339</td>\n",
       "      <td>99773</td>\n",
       "      <td>True</td>\n",
       "      <td>71434</td>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999999</th>\n",
       "      <td>2024-04-29 09:39:45</td>\n",
       "      <td>28340</td>\n",
       "      <td>99774</td>\n",
       "      <td>True</td>\n",
       "      <td>71434</td>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000000 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      data  pedidos_realizados  acumulado_pedidos_realizados  \\\n",
       "0      2022-01-17 08:00:00                   0                             0   \n",
       "1      2022-01-17 08:00:15                   0                             0   \n",
       "2      2022-01-17 08:00:30                   0                             0   \n",
       "3      2022-01-17 08:00:45                   0                             0   \n",
       "4      2022-01-17 08:01:00                   0                             0   \n",
       "...                    ...                 ...                           ...   \n",
       "999995 2024-04-29 09:38:45               28338                         99772   \n",
       "999996 2024-04-29 09:39:00               28339                         99773   \n",
       "999997 2024-04-29 09:39:15               28339                         99773   \n",
       "999998 2024-04-29 09:39:30               28339                         99773   \n",
       "999999 2024-04-29 09:39:45               28340                         99774   \n",
       "\n",
       "        pedido_em_execucao  acumulado_pedidos_finalizados data_abreviada  \\\n",
       "0                    False                              0     2022-01-17   \n",
       "1                    False                              0     2022-01-17   \n",
       "2                    False                              0     2022-01-17   \n",
       "3                    False                              0     2022-01-17   \n",
       "4                    False                              0     2022-01-17   \n",
       "...                    ...                            ...            ...   \n",
       "999995                True                          71434     2024-04-29   \n",
       "999996                True                          71434     2024-04-29   \n",
       "999997                True                          71434     2024-04-29   \n",
       "999998                True                          71434     2024-04-29   \n",
       "999999                True                          71434     2024-04-29   \n",
       "\n",
       "       taxa_de_saida(und/h) takt_time(s/und)  \n",
       "0                 13.556485       203.305085  \n",
       "1                 13.556485       203.305085  \n",
       "2                 13.556485       203.305085  \n",
       "3                 13.556485       203.305085  \n",
       "4                 13.556485       203.305085  \n",
       "...                     ...              ...  \n",
       "999995            16.708861       131.666667  \n",
       "999996            16.708861       131.666667  \n",
       "999997            16.708861       131.666667  \n",
       "999998            16.708861       131.666667  \n",
       "999999            16.708861       131.666667  \n",
       "\n",
       "[1000000 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dados_pedidos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\2638323846.py:1: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  dados_pedidos.loc[:,'taxa_de_saida(und/h)'] = dados_pedidos['taxa_de_saida(und/h)'].fillna(0)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data_abreviada</th>\n",
       "      <th>taxa_de_saida(und/h)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>13.556485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2022-01-18</td>\n",
       "      <td>13.228346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2022-01-19</td>\n",
       "      <td>13.351878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2022-01-20</td>\n",
       "      <td>13.339509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2022-01-21</td>\n",
       "      <td>13.228346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>710</th>\n",
       "      <td>2024-04-24</td>\n",
       "      <td>13.339509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>2024-04-25</td>\n",
       "      <td>13.339509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712</th>\n",
       "      <td>2024-04-26</td>\n",
       "      <td>13.228346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>713</th>\n",
       "      <td>2024-04-27</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>16.708861</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>715 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    data_abreviada taxa_de_saida(und/h)\n",
       "0       2022-01-17            13.556485\n",
       "1       2022-01-18            13.228346\n",
       "2       2022-01-19            13.351878\n",
       "3       2022-01-20            13.339509\n",
       "4       2022-01-21            13.228346\n",
       "..             ...                  ...\n",
       "710     2024-04-24            13.339509\n",
       "711     2024-04-25            13.339509\n",
       "712     2024-04-26            13.228346\n",
       "713     2024-04-27                  0.0\n",
       "714     2024-04-29            16.708861\n",
       "\n",
       "[715 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dados_pedidos.loc[:,'taxa_de_saida(und/h)'] = dados_pedidos['taxa_de_saida(und/h)'].fillna(0)\n",
    "dados_taxa_de_saida = dados_pedidos.groupby(['data_abreviada']).agg({'taxa_de_saida(und/h)': 'mean'})\n",
    "dados_taxa_de_saida = dados_taxa_de_saida.reset_index()\n",
    "dados_taxa_de_saida"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\jhaid\\AppData\\Local\\Temp\\ipykernel_11200\\121579202.py:1: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  dados_pedidos.loc[:,'takt_time(s/und)'] = dados_pedidos['takt_time(s/und)'].fillna(0)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data_abreviada</th>\n",
       "      <th>takt_time(s/und)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2022-01-17</td>\n",
       "      <td>203.305085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2022-01-18</td>\n",
       "      <td>202.40625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2022-01-19</td>\n",
       "      <td>197.469512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2022-01-20</td>\n",
       "      <td>182.966102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2022-01-21</td>\n",
       "      <td>188.284884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>710</th>\n",
       "      <td>2024-04-24</td>\n",
       "      <td>199.907407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>2024-04-25</td>\n",
       "      <td>166.93299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712</th>\n",
       "      <td>2024-04-26</td>\n",
       "      <td>190.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>713</th>\n",
       "      <td>2024-04-27</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>2024-04-29</td>\n",
       "      <td>131.666667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>715 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    data_abreviada takt_time(s/und)\n",
       "0       2022-01-17       203.305085\n",
       "1       2022-01-18        202.40625\n",
       "2       2022-01-19       197.469512\n",
       "3       2022-01-20       182.966102\n",
       "4       2022-01-21       188.284884\n",
       "..             ...              ...\n",
       "710     2024-04-24       199.907407\n",
       "711     2024-04-25        166.93299\n",
       "712     2024-04-26            190.5\n",
       "713     2024-04-27              0.0\n",
       "714     2024-04-29       131.666667\n",
       "\n",
       "[715 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dados_pedidos.loc[:,'takt_time(s/und)'] = dados_pedidos['takt_time(s/und)'].fillna(0)\n",
    "dados_takt_time = dados_pedidos.groupby(['data_abreviada']).agg({'takt_time(s/und)': 'mean'})\n",
    "dados_takt_time = dados_takt_time.reset_index()\n",
    "dados_takt_time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "dados_taxa_de_saida.to_csv('dados_taxa_de_saida.csv', index=False)\n",
    "dados_takt_time.to_csv('dados_takt_time.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
